cmd/tether/attach.go

```diff
@@ -486,34 +500,42 @@ func (t *attachServerSSH) globalMux(reqchan <-chan *ssh.Request) {
 
 		// make sure that errors get send back if we failed
 		if req.WantReply {
-			req.Reply(ok, payload)
+			log.Debugf("Sending global request reply %t back with %#v", ok, payload)
+			if err := req.Reply(ok, payload); err != nil {
+				log.Warnf("Failed to reply a global request back")
+			}
 		}
 
 		// run any pending work now that a reply has been sent
 		if pendingFn != nil {
-			log.Debug("Invoking pending work")
+			log.Debug("Invoking pending work for global mux")
 			go pendingFn()
 			pendingFn = nil
 		}
 	}
 }
 
-func (t *attachServerSSH) channelMux(in <-chan *ssh.Request, session *tether.SessionConfig, detach func()) {
+func (t *attachServerSSH) channelMux(in <-chan *ssh.Request, session *tether.SessionConfig, cleanup func()) {
 	defer trace.End(trace.Begin("attach server channel request handler"))
 
-	// detach
-	defer detach()
+	var ok bool
```

Comment:
Why do you need declaration of these variables here? You could declare them right in place where you need them.